source/elements/oneDAL/source/data_management/data_sources.rst

```diff
@@ -1,7 +1,259 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
 .. _data-sources:
 
 ============
 Data Sources
 ============
 
-TBD
+This section describes the types related to the :txtref:`data-source` concept.
+
+====
+Read
+====
+
+**Read operation** is a function that transforms a data source and other arguments represented via `an args <args_>`_ object
+to `a result <result_>`_ object. An operation responsible for:
```

Comment:
it sits in the same section as csv
In ML there is a part that describes 'anatomy' and parts discussing apis-we can do the same here - find minimum amount for this 'anatomy section' in dm from algorithm anatomy in terms of description by analogy
Re csv we wanted to align with with table section in terms of describing minimal set of requirements we are aware about, table that list possible data sources (today, it is one), etc

Reply:
Partially done.